packages/mdc-menu/adapter.js

```diff
@@ -37,128 +34,88 @@
+   * @param {number} index
+   * @param {string} attr
    */
-  deregisterInteractionHandler(type, handler) {}
-
-  /** @param {function(!Event)} handler */
-  registerBodyClickHandler(handler) {}
-
-  /** @param {function(!Event)} handler */
-  deregisterBodyClickHandler(handler) {}
+  removeAttributeFromElementAtIndex(index, attr) {}
 
   /**
-   * @param {EventTarget} target
-   * @return {number}
+   * Returns true if the element contains the className.
+   * @param {HTMLElement} element
+   * @param {!string} className
+   * @return {boolean} true if the element contains the className
    */
-  getIndexForEventTarget(target) {}
-
-  /** @param {{index: number}} evtData */
-  notifySelected(evtData) {}
-
-  notifyCancel() {}
-
-  saveFocus() {}
+  elementContainsClass(element, className) {}
 
-  restoreFocus() {}
-
-  /** @return {boolean} */
-  isFocused() {}
-
-  focus() {}
-
-  /** @return {number} */
-  getFocusedItemIndex() /* number */ {}
-
-  /** @param {number} index */
-  focusItemAtIndex(index) {}
-
-  /** @return {boolean} */
-  isRtl() {}
-
-  /** @param {string} origin */
-  setTransformOrigin(origin) {}
+  /**
+   * Closes the menu-surface.
+   */
+  closeSurface() {};
 
-  /** @param {{
-  *   top: (string|undefined),
-  *   right: (string|undefined),
-  *   bottom: (string|undefined),
-  *   left: (string|undefined)
-  * }} position */
-  setPosition(position) {}
+  /**
+   * Returns the index for the element provided.
+   * @param {HTMLElement} element
+   * @return {number} index of the element in the list or -1 if it is not in the list.
+   */
+  getElementIndex(element) {};
 
-  /** @param {string} height */
-  setMaxHeight(height) {}
+  /**
+   * Returns the parentElement of the provided element.
+   * @param {HTMLElement} element
+   * @return {?HTMLElement} parentElement of the element provided.
+   */
+  getParentElement(element) {};
 
   /**
-   * @param {number} index
-   * @param {string} attr
-   * @param {string} value
+   * Returns the element within the selectionGroup containing the selected element class.
+   * @param {HTMLElement} selectionGroup
+   * @return {number} element within the selectionGroup that contains the selected element class.
    */
-  setAttrForOptionAtIndex(index, attr, value) {}
+  getSelectedElementIndex(selectionGroup) {};
 
   /**
-   * @param {number} index
-   * @param {string} attr
+   * Emits an event using the evtData.
+   * @param {{
+ *   index: number
+ * }} evtData
    */
-  rmAttrForOptionAtIndex(index, attr) {}
+  notifySelected(evtData) {};
 
   /**
+   * Returns the checkbox contained within the element at the index specified.
    * @param {number} index
-   * @param {string} className
    */
-  addClassForOptionAtIndex(index, className) {}
+  getCheckbox(index) {};
 
   /**
-   * @param {number} index
-   * @param {string} className
+   * Toggles the checkbox within a list item, if any.
+   * @param {HTMLElement} target
```

Comment:
Non-primitive types should be explicitly non-null: `{!HTMLElement}`

Reply:
I wasn't able to remove all non-null types, but I went through and declared everything either nullable/non-nullable.